src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,220 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.StringBuilder;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The `Storage` class represents a tool to store tasks in file and read from file.
+ * It provides methods to store tasks in file and read from file.
+ */
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Constructor of a storage that stores items at a specified file path.
+     *
+     * @param filePath File path to store items.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads stored tasks from the specified file path.
+     *
+     * @return A list of tasks stored in the specified file path.
+     * @throws IllegalArgumentException If error in accessing file to retrieve tasks.
+     */
+    public List<Task> load() throws DukeException {
+        List<Task> items = new ArrayList<>();
+        String home = System.getProperty("user.dir");
+        String[] splitedFile = this.filePath.split("/");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(home);
+
+        // Ensure that the file exist
+        for (int i = 0; i < splitedFile.length; i++) {
+            sb.append("/" + splitedFile[i]);
+            Path path = Paths.get(sb.toString());
+
+            // Folder not exist
+            if (i != splitedFile.length - 1 && !Files.exists(path)) {
+                try {
+                    Files.createDirectory(path);
+                } catch (Exception e) {
+                    throw new DukeException(e.getMessage());
+                }
+            }
+
+            // Folder exist but not folder
+            if (i != splitedFile.length - 1 && !Files.isDirectory(path)) {
+                throw new DukeException();
+            }
+
+            // File not exist
+            if (i == splitedFile.length - 1 && !Files.exists(path)) {
+                try {
+                    Files.createFile(path);
+                } catch (Exception e) {
+                    throw new DukeException(e.getMessage());
+                }
+            }
+
+            // File exist but not exist
+            if (i == splitedFile.length - 1 && Files.isDirectory(path)) {
+                throw new DukeException();
+            }
+        }
+
+        Path path = Paths.get(home, this.filePath);
+
+        // Read from file
+        try {
```

Comment:
Hmm it's quite hard for me to simplify it due to the design of my code. To be more precise:
1. The `try` block is required to catch exceptions.
2. The `while` block is required to read inputs continuously.
3. The `for` block is required to consider every task.
4. The `switch` block is required to decide which command to perform.
5. Some `try` block deep inside is indeed redundant as there is a big `try` block at very outside.
6. Some `if-else` block is required to check for specific condition.

Feel free to tell me if you have any specific suggestion, thanks!

Reply:
1. Perhaps combine multiple `try...catch` blocks into one? I also noticed that you catch `Exception` instead of specifying the specific exception to catch (e.g. `IOException` or `FileNotFoundException`) which is something we what to avoid (I subconsciously remembered in one of the briefing video @damithc mentioned something along the lines of "Don't catch them all" but I can't seem to find it in the textbook). Also, take note to catch all checked exceptions (refer to W1.5 [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week1/topics.html#W1-5)) The following example might also fix your issues.
```java
try {
    <your code>
} catch (IOException e) {
    <your code>
} catch (FileNotFoundException e) {
    <your code>
}
```
2./3./4./5./6. Perhaps the blocks can be abstracted into smaller methods.

Not really sure if mine is a good example but maybe you can take some inspiration [here](https://github.com/SimKianSeng/ip/blob/master/src/main/java/simpli/storage/Storage.java)?

As everyone has different implementations it's quite hard for me to tell you what is right, but I can point out what doesn't seem right to me and maybe provide you with the reference materials. Hope this was helpful.

Reply:
Thanks for your nice elaboration, I will try to apply them in my code!